### PR TITLE
Set proper curl proxy or use Moodle curl class

### PR DESCRIPTION
**Pull request by ottenhoff**
This is completely untested.

### CODE REVIEW

File: classes/task/send_files.php [hunk 1]
```diff
@@ -94,33 +94,28 @@ public function plagiarism_vericite_send_files()
...
+                                $params = array('file' => $filepath);
+                                $options = array('CURLOPT_CONNECTTIMEOUT' => 120);
                                 //Set Encryption from header
                                 if ($externalcontentuploadinfo->getHeaders()) {
                                     $headers = [];
                                     foreach ($externalcontentuploadinfo->getHeaders() as $key => $value) {
                                         $headers[] = "$key: $value";
                                     }
-                                    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
+                                    $options['CURLOPT_HTTPHEADER'] = $headers;
```
Comment (ottenhoff) [inline classes/task/send_files.php]: @baholladay can you change this line to 

$c->setHeader($headers);

Comment (baholladay) [reply classes/task/send_files.php]: Same error:

<Error><Code>InvalidArgument</Code><Message>Only one auth mechanism allowed; only the X-Amz-Algorithm query parameter, Signature query string parameter or the Authorization header should be specified</Message><ArgumentName>Authorization</ArgumentName><ArgumentValue>Basic YW5vbnltb3VzOiBub3JlcGx5QG1vb2RsZS5vcmc=</ArgumentValue><RequestId>B519BDE93F44994E</RequestId><HostId>f0v7eZpxUTbGkssWrCKxgSNih6lp5YnYCPkWKxKkK+PzlE27ybUZUc4MN9HSJ+Imz3hpdRziALk=</HostId></Error>

**Comment by ottenhoff** [comment]
Okay @baholladay updated. Can you test again?

**Comment by baholladay** [comment]
I tested locally with 1 assignment and submission... worked fine!

**Comment by baholladay** [comment]
Is this good to merge?

**Pull request merged**